packages/xstate-wallet/src/channel-wallet.ts

```diff
@@ -235,134 +224,60 @@ export class ChannelWallet {
-    const channel = await this.store.getEntry(channelId);
-    if (!this.registeredChannels.has(channelId)) {
-      this.store.chain.chainUpdatedFeed(channelId).subscribe({
-        next: chainInfo => this.onFundingUpdate(channelId, chainInfo)
-      });
-      this.registeredChannels.add(channelId);
-    }
-    const pk = await this.store.getPrivateKey(await this.store.getAddress());
-    const depositInfo = await this.getDepositInfo(channelId);
-
-    const response = this.crankOpenChannelObjective(
-      channel,
-      this.channelFunding[channelId] ?? {amountOnChain: '0x0'},
-      depositInfo,
-      pk
-    );
-    if (response.deposit) {
-      this.channelFunding[channelId] = {...this.channelFunding[channelId], status: 'DEPOSITED'};
-      await this.store.chain.deposit(
-        channel.channelId,
-        this.channelFunding[channelId].amountOnChain,
-        depositInfo.myDeposit
-      );
-    }
-    if (response.signedStates[0]) {
-      await this.store.addState(response.signedStates[0]);
-      this.messagingService.sendMessageNotification({
-        ...response,
-        walletVersion: '@statechannels/server-wallet@1.23.0)'
-      });
-    }
-    return response;
-  }
-
-  crankOpenChannelObjective(
-    channel: ChannelStoreEntry,
-    channelFunding: Funding,
-    depositInfo: DepositInfo,
-    pk: string
-  ): Response {
-    const response: Response = {
-      objectives: [],
-      signedStates: []
-    };
-    const {latestState} = channel;
-    // Should we sign the prefund state?
-    if (latestState.turnNum === 0 && !channel.isSupportedByMe) {
-      const newState = channel.signAndAdd(latestState, pk);
-      response.signedStates = [newState];
-      return response;
-    }
-
-    // Should we deposit?
-    if (
-      BN.gte(channelFunding.amountOnChain, depositInfo.depositAt) &&
-      BN.lt(channelFunding.amountOnChain, depositInfo.totalAfterDeposit) &&
-      channelFunding.status !== 'DEPOSITED'
-    ) {
-      response.deposit = true;
-      return response;
-    }
-
-    // Should we sign the postfund state?
-    if (
-      BN.gte(channelFunding.amountOnChain, depositInfo.fundedAt) &&
-      latestState.turnNum === 3 &&
-      channel.latestSignedByMe.turnNum === 0
-    ) {
-      const newState = channel.signAndAdd(latestState, pk);
-      response.signedStates = [newState];
-      return response;
-    }
-
-    return response;
-  }
+  /**
+   *  START of wallet 2.0
+   */
 
-  async onFundingUpdate(channelId: string, channelChainInfo: ChannelChainInfo): Promise<void> {
-    this.channelFunding[channelId] = {
-      ...this.channelFunding[channelId],
-      amountOnChain: channelChainInfo.amount
-    };
-    await this.onOpenChannelObjective(channelId);
-  }
+  private async crankRichObjectives(event: DirectFunder.OpenChannelEvent): Promise<void> {
+    const richObjectives = this.store.richObjectives;
+    for (const channelId of Object.keys(richObjectives)) {
```

Comment:
Why should you crank every active objective? I'm imagining that an event should be directed to a specific objective.

Reply:
Good question! It might be that the `crankRichObjectives` signature is not the correct one. It is not clear to me which component is responsible for converting an incoming message or chain event to an objective event.

It could be that `crankRichObjective` accepts a data structure that is **not** an objective event and then tries to match the data structure to ongoing objectives.